src/charm.py

```diff
@@ -234,9 +235,16 @@ def reconfigure_nats(self):
             self.NATS_SERVER_CONFIG_PATH.write_text(rendered_content)
             if self.state.is_started:
                 subprocess.check_call(['systemctl', 'restart', self.NATS_SERVICE])
-        self.unit.status = ActiveStatus()
         self.client.expose_nats(auth_token=self.state.auth_token)
 
+        port = '{}/tcp'.format(self.model.config['client-port'])
```

Comment:
Doesn't this risk an Exception if `client-port` is unset in the configuration ?

Reply:
It should be always set, shouldn't it?

Reply:
I'm not 100% sure how the operator framework handles this, but what would be the behavior if we run `juju config nats client-port=` ? Also we should probably check that the port is a valid one before trying to open some nonsense port like `None/tcp` or the likes

Reply:
A `juju config client-port=` should still not remove the config key from the map. Checking for an empty string or a none value is indeed something we should do.